ember-core/shared/src/test/scala/org/http4s/ember/core/h2/H2StreamSuite.scala

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2019 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.ember.core.h2
+
+import cats.effect.Deferred
+import cats.effect.IO
+import cats.effect.Ref
+import cats.effect.std.Queue
+import cats.syntax.all._
+import fs2.Chunk
+import org.http4s.Headers
+import org.http4s.Http4sSuite
+import org.http4s.HttpVersion
+import org.http4s.Request
+import org.http4s.Response
+import org.http4s.Status
+import org.typelevel.log4cats
+import scodec.bits.ByteVector
+
+class H2StreamSuite extends Http4sSuite {
+  val defaultSettings = H2Frame.Settings.ConnectionSettings.default
+
+  def streamAndQueue(
+      config: H2Frame.Settings.ConnectionSettings
+  ): IO[(H2Stream[IO], Queue[IO, Chunk[H2Frame]])] =
+    for {
+      writeBlock <- Deferred[IO, Either[Throwable, Unit]]
+      req <- Deferred[IO, Either[Throwable, Request[fs2.Pure]]]
+      resp <- Deferred[IO, Either[Throwable, Response[fs2.Pure]]]
+      trailers <- Deferred[IO, Either[Throwable, Headers]]
+      readBuffer <- Queue.unbounded[IO, Either[Throwable, ByteVector]]
+
+      _ <- writeBlock.complete(Either.unit)
+      _ <- req.complete(Left(new Exception()))
+      _ <- resp.complete(Left(new Exception()))
+      _ <- trailers.complete(Right(Headers.empty))
```

Comment:
Out of curiosity is completing all of these `Deferred`s necessary for the tests to run?

Reply:
Nice catch, no, completing these was not necessary.